huaweicloud/services/dataarts/resource_huaweicloud_dataarts_factory_script.go

```diff
@@ -0,0 +1,307 @@
+		d.Set("configuration", utils.PathSearch("configuration", getRespBody, nil)),
+		d.Set("created_by", utils.PathSearch("createUser", getRespBody, nil)),
+		d.Set("auto_acquire_lock", utils.PathSearch("autoAcquireLock", getRespBody, nil)),
+	)
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceStudioScriptUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		httpUrl = "v1/{project_id}/scripts/{script_name}"
+		product = "dataarts-dlf"
+	)
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio client: %s", err)
+	}
+
+	updatePath := client.Endpoint + httpUrl
+	updatePath = strings.ReplaceAll(updatePath, "{project_id}", client.ProjectID)
+	updatePath = strings.ReplaceAll(updatePath, "{script_name}", d.Id())
+
+	updateOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"workspace": d.Get("workspace_id").(string)},
+		JSONBody:         utils.RemoveNil(buildCreateOrUpdateScriptBodyParams(d)),
+		OkCodes:          []int{204},
+	}
+
+	_, err = client.Request("PUT", updatePath, &updateOpt)
+	if err != nil {
+		return diag.Errorf("error updating DataArts script: %s", err)
+	}
+
+	return resourceStudioScriptRead(ctx, d, meta)
+}
+
+func resourceStudioScriptDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		httpUrl = "v1/{project_id}/scripts/{script_name}"
+		product = "dataarts-dlf"
+	)
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio client: %s", err)
+	}
+
+	deletePath := client.Endpoint + httpUrl
+	deletePath = strings.ReplaceAll(deletePath, "{project_id}", client.ProjectID)
+	deletePath = strings.ReplaceAll(deletePath, "{script_name}", d.Id())
+	deleteOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"workspace": d.Get("workspace_id").(string)},
+	}
+
+	_, err = client.Request("DELETE", deletePath, &deleteOpt)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func resourceStudioScriptImportState(_ context.Context, d *schema.ResourceData,
+	_ interface{}) ([]*schema.ResourceData, error) {
+	parts := strings.Split(d.Id(), "/")
+	partLength := len(parts)
+
+	if partLength == 2 {
+		d.SetId(parts[1])
+		return []*schema.ResourceData{d}, d.Set("workspace_id", parts[0])
+	}
+	return nil, fmt.Errorf("invalid format specified for import id, must be <workspace_id>/<name>")
+}
+
+func buildCreateOrUpdateScriptBodyParams(d *schema.ResourceData) map[string]interface{} {
+	return map[string]interface{}{
+		"name":           d.Get("name"),
+		"type":           d.Get("type"),
+		"directory":      d.Get("directory"),
```

Comment:
should use utils.ValueIgnoreEmpty for optional parameters
```suggestion
		"directory":      utils.ValueIgnoreEmptyd.Get("directory")),
```